File: ditto-chat-client/src/app/interfaces/ChatThreadDto.ts
import ChatThreadOverviewDto from "./ChatThreadOverviewDto";
import ChatThreadMessageDto from "./ChatThreadMessageDto";
import PagedListDto from "./PagedListDto";

export default interface ChatThreadDto {
    chatThreadOverview: ChatThreadOverviewDto;
    chatThreadMessages: PagedListDto<ChatThreadMessageDto>;
}

File: ditto-chat-server/src/main/java/com/ditto_chat/ditto_chat_server/dtos/ChatThreadDto.java
package com.ditto_chat.ditto_chat_server.dtos;

public class ChatThreadDto {
    private ChatThreadOverviewDto chatThreadOverview;
    private ResponsePagedListDto<ChatThreadMessageDto> chatThreadMessages;
    
    public ChatThreadDto(ChatThreadOverviewDto chatThreadOverview, ResponsePagedListDto<ChatThreadMessageDto> chatThreadMessages) {
        this.chatThreadOverview = chatThreadOverview;
        this.chatThreadMessages = chatThreadMessages;
    }

    public ChatThreadOverviewDto getChatThreadOverview() {
        return chatThreadOverview;
    }

    public ResponsePagedListDto<ChatThreadMessageDto> getChatThreadMessages() {
        return chatThreadMessages;
    }
}
